frontend/public/components/pod-logs.jsx

```diff
@@ -1,194 +1,79 @@
-    this._endStreaming();
-    this._buffer = lineBuffer(1000);
-    const currentContainer = this.state.containerNames[index];
-    this.setState({
-      currentContainer,
-      logURL: this._logURL(this.props.obj, currentContainer),
-      logState: 'loading'
-    }, this._beginStreaming);
-  }
-
-  _updateLogState(newState) {
-    this.setState({
-      logState: newState
-    });
-  }
-
-  _toggleLogState() {
-    this.setState({
-      logState: this.state.logState === 'streaming' ? 'paused' : 'streaming'
-    });
-  }
-
-  _touchLoadTime() {
-    this._touchLoadTimeState();
-    this._loadTime = Date.now();
-  }
-
-  // separate function so that it can be throttled
-  _touchLoadTimeState() {
-    this.setState({
-      loadTime: this.state.loadTime + 1
-    });
-  }
-
-  _endStreaming() {
-    this._stream.abort();
-    clearTimeout(this._pendingReload);
-  }
-
-  _resetPendingReload() {
-    const sinceLastLoad = Date.now() - this._loadTime;
-    const wait = Math.max(0, (1000 * 5) - sinceLastLoad);
-    this._pendingReload = setTimeout(() => {
-      if (this.state.logState === 'paused') {
-        // don't reset stream if the user paused the stream
-        this._touchLoadTime();
-        this._resetPendingReload();
-        return;
-      }
-      this._beginStreaming();
-    }, wait);
-  }
+    if (_.isEmpty(prevState.currentContainer) && newState.containers.length > 0) {
+      newState.currentContainer = newState.containers[0];
+    }
 
-  _loadStarted() {
-    this.setState({
-      logState: 'streaming'
-    });
-  }
+    const currentContainer = newState.currentContainer || prevState.currentContainer;
+    newState.eof = !containerIsRunning(currentContainer);
 
-  _processData(data) {
-    if (dataHasHTML(data)) {
-      this._buffer.push('Logs are currently unavailable');
-    } else if (!dataHasFailureMsg(data)) {
-      this._buffer.push(data);
+    if( !_.isEqual(prevState.currentContainer, newState.currentContainer)
+        || !_.isEqual(prevState.containers, newState.containers)
+        || prevState.eof !== newState.eof) {
+      return newState;
     }
-
-    this._touchLoadTime();
+    return null;
   }
 
-  _beginStreaming() {
-    clearTimeout(this._pendingReload);
-    this._pendingReload = null;
-    this._buffer = lineBuffer(1000);
-
-    this._stream = stream(this.state.logURL, this._loadStarted.bind(this), this._processData.bind(this));
-    this._stream.promise
-      .then(() => !this._pendingReload && this._resetPendingReload()) // Load ended
-      .catch((why) => { // Load failed/aborted
-        if (why === 'abort') {
-          return;
-        }
-
-        this._loadStarted();
-        this._buffer.push(`Error: ${why}`);
-        this._touchLoadTime();
-        if (!this._pendingReload) {
-          this._resetPendingReload();
-        }
-      });
+  _selectContainer(index) {
+    const currentContainer = this.state.containers[index];
+    const eof = !containerIsRunning(currentContainer);
```

Comment:
It is unexpected that selecting a container would have this side effect because `eof` is a property of the container.  If you change `status` above, you can clean up this code too.